### ISSUE TITLE
Windows CMake+Ninja build warning due to duplicate basename

### ISSUE THREAD

**Issue by seanmiddleditch** [microsoft/mimalloc#6]
Building a monolithic project with CMake+Ninja (default behaviour in Visual Studio these days) that adds mimalloc via `add_subdirectory`:

```
ninja : warning : multiple rules generate lib/mimalloc-debug.lib. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
```

The problem appears to be that both the shared and static versions of the library end up creating a `mimalloc-static.lib` file (the import library for the DLL and the static library itself). This is because both libraries set:

```
set_target_properties(mimalloc-static PROPERTIES OUTPUT_NAME ${mi_basename})
```

Which causes both the DLL's import library and the static library's name to be identical.

```
#############################################
# Link the shared library bin\mimalloc-debug.dll

build bin\mimalloc-debug.dll lib\mimalloc-debug.lib: C_SHARED_LIBRARY_LINKER__mimalloc
  ...stuff...
  TARGET_FILE = bin\mimalloc-debug.dll
  TARGET_IMPLIB = lib\mimalloc-debug.lib
  TARGET_PDB = bin\mimalloc-debug.pdb
```

```
#############################################
# Link the static library lib\mimalloc-debug.lib

build lib\mimalloc-debug.lib: C_STATIC_LIBRARY_LINKER__mimalloc-static
  ...stuff...
  TARGET_FILE = lib\mimalloc-debug.lib
  TARGET_PDB = lib\mimalloc-debug.pdb
```

**Comment by daanx** [comment]
Great analysis. Thanks! What would you recommend to change? Add "-imp" to the import library name or would that cause trouble at link time? Perhaps we should just not even build the static library as we already create a single static object file too.

**Comment by daanx** [comment]
I think this is fixed now by #22
